bson/src/main/org/bson/LazyBSONObject.java

```diff
@@ -177,7 +177,7 @@ Object readValue(final BsonBinaryReader reader) {
             case BINARY:
                 byte binarySubType = reader.peekBinarySubType();
                 if (BsonBinarySubType.isUuid(binarySubType) && reader.peekBinarySize() == 16) {
-                    return new UuidCodec().decode(reader, DecoderContext.builder().build());
+                    return new UuidCodec(UuidRepresentation.JAVA_LEGACY).decode(reader, DecoderContext.builder().build());
```

Comment:
This could wait until 4.x (since in 3.x JAVA_LEGACY is the default anyway) , but it doesn't hurt and I wanted to call attention to it.

Reply:
Ack